writer/ch_wrapper/http_client.go

```diff
@@ -0,0 +1,409 @@
+package ch_wrapper
+
+import (
+	"context"
+	"fmt"
+	"github.com/ClickHouse/ch-go"
+	"github.com/ClickHouse/ch-go/proto"
+	"github.com/metrico/qryn/writer/utils/heputils"
+	"github.com/metrico/qryn/writer/utils/logger"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+)
+
+// HttpChClient implements the IChClient interface for HTTP connections
+type HttpChClient struct {
+	conn clickhouse.Conn
+}
+
+func NewHttpChClientFactory(dsn string) IChClientFactory {
+	return func() (IChClient, error) {
+		options, err := clickhouse.ParseDSN(dsn)
+		if err != nil {
+			options = &clickhouse.Options{
+				Addr: []string{dsn},
+				Auth: clickhouse.Auth{
+					Database: "default",
+					Username: "default",
+					Password: "",
+				},
+				Protocol: clickhouse.HTTP,
+				Settings: clickhouse.Settings{
+					"max_execution_time": 60,
+				},
+				DialTimeout:      time.Second * 30,
+				MaxOpenConns:     10,
+				MaxIdleConns:     5,
+				ConnMaxLifetime:  time.Hour,
+				ConnOpenStrategy: clickhouse.ConnOpenInOrder,
+			}
+		} else {
+			options.Protocol = clickhouse.HTTP
+		}
+
+		conn, err := clickhouse.Open(options)
```

Comment:
According to https://github.com/ClickHouse/clickhouse-go?tab=readme-ov-file#http-support-experimental the http access is only supported via `clickhouse.OpenDB`, not `clickhouse.Open`. Please use `clickhouse.OpenDB` here.